ctapipe/tools/make_irf.py

```diff
@@ -0,0 +1,385 @@
+    ).tag(config=True)
+
+    chunk_size = Integer(
+        default_value=100000,
+        allow_none=True,
+        help="How many subarray events to load at once for making predictions.",
+    ).tag(config=True)
+
+    output_path = traits.Path(
+        default_value="./IRF.fits.gz",
+        allow_none=False,
+        directory_ok=False,
+        help="Output file",
+    ).tag(config=True)
+
+    overwrite = Bool(
+        False,
+        help="Overwrite the output file if it exists",
+    ).tag(config=True)
+
+    obs_time = Float(default_value=50.0, help="Observation time").tag(config=True)
+    obs_time_unit = Unicode(
+        default_value="hour",
+        help="Unit used to specify observation time as an astropy unit string.",
+    ).tag(config=True)
+
+    alpha = Float(
+        default_value=0.2, help="Ratio between size of on and off regions"
+    ).tag(config=True)
+    ON_radius = Float(default_value=1.0, help="Radius of ON region in degrees").tag(
+        config=True
+    )
+    max_bg_radius = Float(
+        default_value=3.0, help="Radius used to calculate background rate in degrees"
+    ).tag(config=True)
+
+    classes = [CutOptimising, DataBinning, OutputEnergyBinning, EventPreProcessor]
+
+    def make_derived_columns(self, kind, events, spectrum, target_spectrum, obs_conf):
+
+        if obs_conf["subarray_pointing_lat"].std() < 1e-3:
+            assert all(obs_conf["subarray_pointing_frame"] == 0)
+            # Lets suppose 0 means ALTAZ
+            events["pointing_alt"] = obs_conf["subarray_pointing_lat"][0] * u.deg
+            events["pointing_az"] = obs_conf["subarray_pointing_lon"][0] * u.deg
+        else:
+            raise NotImplementedError(
+                "No support for making irfs from varying pointings yet"
+            )
+
+        events["theta"] = calculate_theta(
+            events,
+            assumed_source_az=events["true_az"],
+            assumed_source_alt=events["true_alt"],
+        )
+        events["true_source_fov_offset"] = calculate_source_fov_offset(
+            events, prefix="true"
+        )
+        events["reco_source_fov_offset"] = calculate_source_fov_offset(
+            events, prefix="reco"
+        )
+        # Gamma source is assumed to be pointlike
+        if kind == "gamma":
+            spectrum = spectrum.integrate_cone(0 * u.deg, self.ON_radius * u.deg)
+        events["weight"] = calculate_event_weights(
+            events["true_energy"],
+            target_spectrum=target_spectrum,
+            simulated_spectrum=spectrum,
+        )
+
+        return events
+
+    def get_metadata(self, loader):
+        obs = loader.read_observation_information()
+        sim = loader.read_simulation_configuration()
+
+        # These sims better have the same viewcone!
+        assert sim["max_viewcone_radius"].std() == 0
+        sim_info = SimulatedEventsInfo(
+            n_showers=sum(sim["n_showers"] * sim["shower_reuse"]),
```

Comment:
Here there is an implicit assumption that all of the runs contained in the file use the same distribution of thrown events (in energy and impact).  However, that is not necessary to assume.  We can easily support having mixed ranges, and indeed that will likely happen in the future (e.g. to fill in statistics at high energy).  

So at the very least this assumption should be made explicit and you should have a sanity check here that they are all the same, and raise an exception if not.  At the very least, add some more asserts, but better raise a more useful exception explaining to the user that the inputs don't agree and this tool assumes they do.

Later, we should really use the actual thrown energy histograms (`/simulation/service/shower_distribution`)! Those are in the file and can also be summed up to produce the final histogram. In that case, no PowerLaws need to be created, as the shape can be anything.

Reply:
> Here there is an implicit assumption that all of the runs contained in the file use the same distribution of thrown events (in energy and impact). However, that is not necessary to assume. We can easily support having mixed ranges, and indeed that will likely happen in the future (e.g. to fill in statistics at high energy).

This is at the moment not supported by pyirf effective area computation, you need to pass a single of these `SimulatedEventsInfo` objects.

Agree that a check should be done though that `e_min`, `e_max` and `spectral_index` are indentical.

Reply:
> Later, we should really use the actual thrown energy histograms (/simulation/service/shower_distribution)! 

This is what we should already do now! The total number of events should come from the shower histograms, not the header information.

Reply:
Added reading of the shower distribution and using that for the number of showers, changed the max viewcone assert for three if statments checking e_min, e_max and spectral_index.

Reply:
Great. Even if we don't support using the actual thrown histogram yet in pyirf, using them to get the counts is still probably safer as we do not write those histograms unless the full file is processed (_max-events_ is not set). Though I suppose in that case we would then potentially undercount instead of overcount when summing them...